syntax = "proto3";

package agentcore;

service AgentCore {
  // Run inference with input data
  rpc Inference (InferenceRequest) returns (InferenceResponse);
  // Get agent status
  rpc Status (StatusRequest) returns (StatusResponse);
  // Hot-swap agent core (reload, upgrade, etc)
  rpc HotSwap (HotSwapRequest) returns (HotSwapResponse);
}

message InferenceRequest {
  string input = 1;
}

message InferenceResponse {
  string output = 1;
  string agent_version = 2;
}

message StatusRequest {}
message StatusResponse {
  string agent_version = 1;
  string status = 2;
  string uptime = 3;
}

message HotSwapRequest {
  string new_image = 1;
}
message HotSwapResponse {
  bool success = 1;
  string message = 2;
}
